### DISCUSSION TITLE
wp_nav_menu - walker

### DISCUSSION THREAD

**Discussion by ant31zac** [bootscore/bootscore#493] [Q&A]
Hello everyone,
I would like to know how can I overwrite the current bootscoore nav_walker?

thanks in advance to everyone who will waste time to answer my question

**Comment by crftwrk** [comment]
Hi,

check this https://github.com/orgs/bootscore/discussions/347. Here is explained how to register nav_walker in your child and add more levels depth.

**Reply by finalwebsites** [reply]
Hi,

I followed this steps from the provided link. The information was a bit outdated, but I was able to fix it.
In my child theme I have my own version of the walker class and the main-menu.php template part is available and modified.
The depth paramater is still not working, the items with additional sub-items doesn't show up. 
To check if the modified version is walking, I checked it while adding another css-class value.

Why do I need to use a copy from that external walker class script and not the version which is included in the main theme?

**Reply by finalwebsites** [reply]
... I found it already, in the current version you used a prefix for the function register_navwalker. It's now bootscore_register_navwalker

Still curious why I need to use my own walker class :)

**Reply by crftwrk** [reply]
Yes, link above is for v5. In v6 nav-walker have been moved into a pluggable function https://github.com/bootscore/bootscore/blob/main/inc/navwalker.php.

> Still curious why I need to use my own walker class :)

What do you mean here?
